provider/spark.go

```diff
@@ -684,10 +687,49 @@ func (s *SparkGenericExecutor) InitializeExecutor(store SparkFileStore) error {
 	return nil
 }
 
+func (s *SparkGenericExecutor) getYarnCommand(args string) (string, error) {
+	configDir, err := os.MkdirTemp("", "hadoop-conf")
+	if err != nil {
+		return "", fmt.Errorf("could not create temp dir: %v", err)
+	}
+	coreSitePath := filepath.Join(configDir, "core-site.xml")
+	yarnSitePath := filepath.Join(configDir, "yarn-site.xml")
```

Comment:
small thing but might be better to pair this with the write below.

Reply:
fixed